sentry_sdk/integrations/pymongo.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import absolute_import
+
+from sentry_sdk import Hub
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.tracing import Span
+from sentry_sdk.utils import capture_internal_exceptions
+
+from sentry_sdk._types import MYPY
+
+try:
+    from pymongo import monitoring
+except ImportError:
+    raise DidNotEnable("Pymongo not installed")
+
+if MYPY:
+    from typing import Dict
+
+    from pymongo.monitoring import (
+        CommandFailedEvent,
+        CommandStartedEvent,
+        CommandSucceededEvent,
+    )
+
+
+class CommandTracer(monitoring.CommandListener):
+    def __init__(self):
+        self._ongoing_operations = {}  # type: Dict[int, Span]
+
+    def _operation_key(self, event):
+        # type: (CommandFailedEvent | CommandStartedEvent | CommandSucceededEvent) -> int
+        return event.request_id
+
+    def started(self, event):
+        # type: (CommandStartedEvent) -> None
+        hub = Hub.current
+        if hub.get_integration(PyMongoIntegration) is None:
+            return
+        with capture_internal_exceptions():
+            command = dict(event.command)
+
+            command.pop("$db", None)
+            command.pop("$clusterTime", None)
+            command.pop("$signature", None)
+
+            if event.command_name:
+                op = "db." + event.command_name
```

Comment:
As the span `op` please use "db.query" instead of "db.find". (I know it messes up this elegant code but it helps us figure out performance problems with queries)

Reply:
Nah, it's fine and actually solves a discussion above if `command_name` makes sense being duplicated in `op` and `description`.

Reply:
We should try to stay in sync with the list of operations outlined at https://develop.sentry.dev/sdk/performance/span-operations/#database
We just spend a significant amount of time unifying span ops across our SDKs.

Reply:
Yeah, I've seen that change. I believe this doc looked a bit different when I was writing the pull request. 😄 Now it's a bit cleaner.